odoo/fields.py

```diff
@@ -2376,6 +2376,32 @@ def __init__(self, comodel_name=Default, relation=Default, column1=Default,
             **kwargs
         )
 
+    def _get_mro_has_field_relation(self, model, name):
+        for mro in type(model).__mro__:
+            if hasattr(mro, '_abstract') and\
+                    not mro._abstract and name in mro.__dict__:
+                field = mro.__dict__[name]
+                return self.relation == field.args.get('relation')
+        return False
+
+    def _check_relation(self, model, name):
+        if isinstance(model._inherit, str):
+            inheriteds = [model._inherit]
+        elif isinstance(model._inherit, list):
+            inheriteds = model._inherit
+        else:
+            return
+        for inherited in inheriteds:
+            inherited_model = model.env[inherited]
+            if model._name != inherited:
+                if self._get_mro_has_field_relation(inherited_model, name):
+                    self.relation = False
+
+    def _setup_attrs(self, model, name):
+        super(Many2many, self)._setup_attrs(model, name)
+        if self.store:
```

Comment:
The error occurs only on stored field as it is related to the relation table argument

Reply:
No, I have tested in other non stored cases and also happens the bug.

For example, in https://github.com/OCA/knowledge/tree/9.0/document_page module, <code>document.page</code> has <code>_inherit = ['mail.thread']</code>, and thus the following breaks
```python
self.env["document.page"].search([('message_partner_ids', '=', 1)])
```
although <code>message_partner_ids</code> is non stored field because is a computed field.